### PR TITLE
Add geometry distance operator

### PR DESCRIPTION
**Pull request by tom-s-powell**
Adding support for PostGIS geometry distance operators:
* https://postgis.net/docs/geometry_distance_knn.html
* https://postgis.net/docs/geometry_distance_box.html

### CODE REVIEW

File: src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt [hunk 1]
```diff
@@ -3110,6 +3110,8 @@ Expression RegularCondition() #RegularCondition:
         | <OP_CONCAT> { result = new JsonOperator("||"); }
         | "-" { result = new JsonOperator("-"); }
         | "-#" { result = new JsonOperator("-#"); }
+        | <-> { result = new GeometryDistance("<->"); }
```
Comment (wumpz) [inline src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt:3113]: How did you test this line of code? It is definitely wrong. I corrected it.

File: src/main/java/net/sf/jsqlparser/util/validation/validator/ExpressionValidator.java [hunk 2]
```diff
@@ -586,4 +587,8 @@ public void visit(IsDistinctExpression isDistinctExpression) {
         isDistinctExpression.getRightExpression().accept(this);
     }
 
+    @Override
+    public void visit(GeometryDistance geometryDistance) {
+        visitOldOracleJoinBinaryExpression(geometryDistance, " <-> ");
```
Comment (wumpz) [inline src/main/java/net/sf/jsqlparser/util/validation/validator/ExpressionValidator.java:592]: This one deparses `<#>` to `<->`. I corrected it.

**Review by wumpz — COMMENTED** [review]
I merged it and corrected the problems. Did you forget to push some commits? I hope this does not offend you, but those problems needed to be addressed.

**Comment by wumpz** [comment]
Could you provide an improvement to allow only  <-> and <#> as operators. At the moment one could misuse `new GeometryDistance("myop")`.

**Comment by tom-s-powell** [comment]
Thanks for the fixes @wumpz. Is there an ETA on a release that'll include this?

**Comment by wumpz** [comment]
Not yet. Still, there are some issues there. Maybe I will speed up and release earlier.

**Comment by wumpz** [comment]
Now ... I delayed some open issues.

**Pull request merged**